sdk/keyvault/azure-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/models/SecretProperties.java

```diff
@@ -130,9 +139,9 @@ public Boolean isEnabled() {
      * Set the enabled value.
      *
      * @param enabled The enabled value to set
-     * @return the SecretBase object itself.
+     * @return the SecretProperties object itself.
      */
-    public SecretBase setEnabled(Boolean enabled) {
+    public SecretProperties setEnabled(Boolean enabled) {
         this.enabled = enabled;
```

Comment:
should we have Objects.requireNonNull in these?

Reply:
Doesn't break the service or code flow.
But worth adding the check for user code readability.

Reply:
added the check